src/utils.js

```diff
@@ -0,0 +1,95 @@
+/* Various Utility Functions */
+
+/* array: array to deep flatten, preserving order of elements
+ *     All elements in sub-arrays are placed in the top level
+ *     Arrays wrapped inside objects are not altered
+ *     TO DO, POSSIBLY: Place this and other functions (scrubObject, scrubArray from defaultOptions.js) in utils.js
+ */
+export function deepFlattenArray(passedArray) {
```

Comment:
Sure. _flattenDeep it is. It looks more robust than what I've written.

The other two are for `defaultOptions.js`, on the idea that, since the options are being passed in from outside the engine, we can't assume they'll be safe. I was concerned about people passing in function as object properties or--now that I'm thinking about it--hiding a payload inside a getter function assigned to an object property.

They're in the interest of whitelisting external input. If that's not needed, I'd be happy to remove them. `_.flattenDeep` and https://www.npmjs.com/package/flat seem like they'd suffice in that case.

If we need to keep this, hmm...I'm looking through lodash documentation right now. I haven't found anything yet, beyond `_.compact`. Do you have any suggestions?